languages/rust/exercises/concept/enums/src/lib.rs

```diff
@@ -0,0 +1,9 @@
+// TODO define enum variants
+pub enum LogLevel {}
+
+pub fn parse_log_level(log: &str) -> LogLevel {
```

Comment:
```suggestion
pub fn parse_log_level(log: &str) -> Option<LogLevel> {
```

We should demonstrate correct practice for the case that the input string doesn't match the regex `^\[(?P<level>[^\]]*)\]:\s*(?P<message>.*)$`.

The difference between `None` and `Some(Unknown(_))`:

`[foo]: a message foo` -> `Some(Unknown("foo"))`
`not a structured log line` -> `None`

Reply:
So if this is added, we'll need a prereq adding for Option types and an exercise around that, or this exercise to teach Option Types and have that in the different README files

Reply:
That's a circular dependency: Option type are enums.

Reply:
OK, cool. So this exercise should probably teach Option types then, and have it added to what it teaches etc :)